pkg/image/generated/internalclientset/typed/image/internalversion/imagestreamimport_expansion.go

```diff
@@ -0,0 +1,26 @@
+package internalversion
+
+import (
+	"time"
+
+	image "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+type ImageStreamImportExpansion interface {
+	CreateWithoutTimeout(*image.ImageStreamImport) (*image.ImageStreamImport, error)
+}
+
+// CreateWithoutTimeout imports the provided images and won't time out after 30 seconds. Use this when you must
+// import a large number of images.
+func (c *imageStreamImports) CreateWithoutTimeout(imageStreamImport *image.ImageStreamImport) (result *image.ImageStreamImport, err error) {
+	result = &image.ImageStreamImport{}
+	err = c.client.Post().
+		Namespace(c.ns).
+		Resource("imagestreamimports").
+		Body(imageStreamImport).
+		// this instructs the api server to allow our request to take up to an hour - chosen as a high boundary
+		Timeout(time.Hour).
```

Comment:
why not just make this an argument to the function?  especially since the function name is currently a lie.

also where's the generator code change that yields this?

Reply:
Because we don't want end users setting arbitrary timeouts.  We define what "no timeout" means behind the interface.  End clients don't get to set it.

Reply:
There is no generator code change.  This is an expansion, which are side car methods that you write and get slotted in to the interfaces.

Reply:
> can't we just call the client directly with Timeout in that place?

the client doesn't expose the timeout field that i saw, so no.